### DISCUSSION TITLE
Multiple sockets for gpg-agent

### DISCUSSION THREAD

**Discussion by high1** [davmac314/dinit#70] [General]
I've found [this](https://forum.artixlinux.org/index.php/topic,3268.0.html) guide on Artix Linux forum, where it's shown how to run gpg-agent in supervised mode:
```
type=process
socket-listen=/run/user/1000/gnupg/S.gpg-agent
command=gpg-agent --supervised
```
and that works well, but this way it only listens on standard (std) socket. Is there a possibility to have multiple socket for service, and to have them named?
Systemd has sockets added like this to the base service:
https://fossies.org/linux/gnupg/doc/examples/systemd-user/gpg-agent.socket
https://fossies.org/linux/gnupg/doc/examples/systemd-user/gpg-agent-ssh.socket
...
something like:
```
 [Socket]
    ListenStream=%t/gnupg/S.gpg-agent
    FileDescriptorName=std
    SocketMode=0600
    DirectoryMode=0700
```
   
```
    [Socket]
    ListenStream=%t/gnupg/S.gpg-agent.ssh
    FileDescriptorName=ssh
    Service=gpg-agent.service
    SocketMode=0600
   DirectoryMode=0700
```
   
   Is this possible with dinit? I need it to listen to the ssh socket also, but I'm clueless on how to add multiple sockets to a service, and also have something like FileDescriptor - gpg-agent expects std, ssh, browser and extra file descriptors.

**Comment by davmac314** [comment]
It isn't currently possible to configure more than one socket per service. For the moment I suggest instead using `type = bgprocess` and `command=gpg-agent --daemon`. (Since it doesn't appear that gpg-agent produces a pid file, I'm afraid dinit won't be able to supervise the process).
